### PR TITLE
[camport3] add new port

### PR DESCRIPTION
**Pull request by syp1975**
**Describe the pull request**

- What does your PR fix?
Adds a new port of https://github.com/percipioxyz/camport3, an sdk to grab images and clouds of points from percipio.xyz's 3d cameras.

- Which triplets are supported/not supported? Have you updated the CI baseline?
supports x64-linux, x86-windows and x64-windows (dynamic linkage only)

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes. SDK contains runtime binaries, include files, docs and sample source code. It does not contains full source code of the library. I'm not sure if this goes against any rule.

### CODE REVIEW

File: ports/camport3/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,67 @@
+include(vcpkg_common_functions)
```
Comment (NancyLi1013) [inline ports/camport3/portfile.cmake:1]:
```suggestion
```
This function is no longer needed here. So please remove this line.

**Review by NancyLi1013 — COMMENTED** [review]
It seems that there are no debug binaries on Windows, which will affect the use on Windows.

**Comment by syp1975** [comment]
Indeed the maker of the SDK does not provides debug binaries, neither for Windows or Linux. At least they provide good code samples.

File: ports/camport3/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+file(MAKE_DIRECTORY ${CURRENT_PACKAGES_DIR}/include)
+file(COPY 
+    ${SOURCE_PATH}/include/TYApi.h
+    ${SOURCE_PATH}/include/TYCoordinateMapper.h
+    ${SOURCE_PATH}/include/TYImageProc.h
+    ${SOURCE_PATH}/include/TyIsp.h
+    DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+
+if(NOT VCPKG_CMAKE_SYSTEM_NAME OR VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore")
```
Comment (NancyLi1013) [inline ports/camport3/portfile.cmake:18]:
```suggestion
if(VCPKG_TARGET_IS_WINDPWS)
```

File: ports/camport3/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+            ${SOURCE_PATH}/lib/win/hostapp/${VCPKG_TARGET_ARCHITECTURE}/tycam.lib
+            DESTINATION ${CURRENT_PACKAGES_DIR}/debug/lib
+        )
+        file(COPY
+            ${SOURCE_PATH}/lib/win/hostapp/${VCPKG_TARGET_ARCHITECTURE}/tycam.dll
+            DESTINATION ${CURRENT_PACKAGES_DIR}/debug/bin
+        )
+    endif()
+
+elseif(VCPKG_CMAKE_SYSTEM_NAME STREQUAL "Linux")
```
Comment (NancyLi1013) [inline ports/camport3/portfile.cmake:42]:
```suggestion
elseif(VCPKG_TARGET_IS_LINUX)
```

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

**Comment by NancyLi1013** [comment]
Does this port not support OSX?
If so, please add `Supports: !osx` to CONTROL file and also add the unsupported message `vcpkg_fail_port_install(ON_TARGET "osx" )` at the top of portfile.cmake.

File: ports/camport3/CONTROL [hunk 4]
```diff
@@ -0,0 +1,6 @@
+Source: camport3
+Version: 1.5.3
+Description: percipio.xyz cameras SDK
+Homepage: https://github.com/percipioxyz/camport3
+Build-Depends: libusb, opencv3
```
Comment (ras0219-msft) [inline ports/camport3/CONTROL:5]: Based on the documentation, I believe these are required only for the samples and aren't requirements of this port.
```suggestion
```

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

**Comment by NancyLi1013** [comment]
@syp1975 
Could you please address the review suggestions?

**Comment by syp1975** [comment]
@NancyLi1013 all requested changes resolved now. Thanks for your help.

**Comment by NancyLi1013** [comment]
The failures on x64-windows-static:

```
DLLs should not be present in a static build, but the following DLLs were found:

    D:/packages/camport3_x64-windows-static/bin/tycam.dll
    D:/packages/camport3_x64-windows-static/debug/bin/tycam.dll

There should be no bin\ directory in a static build, but D:\packages\camport3_x64-windows-static\bin is present.
There should be no debug\bin\ directory in a static build, but D:\packages\camport3_x64-windows-static\debug\bin is present.
If the creation of bin\ and/or debug\bin\ cannot be disabled, use this in the portfile to remove them

    if(VCPKG_LIBRARY_LINKAGE STREQUAL "static")
        file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
    endif()

Found 2 error(s). Please correct the portfile:
    C:\a\1\s\ports\camport3\portfile.cmake
```
The failures on arm64-windows:

```
CMake Error at ports/camport3/portfile.cmake:21 (file):
  file COPY cannot find
  "D:/buildtrees/camport3/src/v1.5.3-d74d139b2a.clean/lib/win/hostapp/arm64/tycam.lib":
  File exists.
Call Stack (most recent call first):
  scripts/ports.cmake:79 (include)
```

The failures on x64-uwp:

```
The following DLLs do not have the App Container bit set:

    D:/packages/camport3_x64-uwp/debug/bin/tycam.dll
    D:/packages/camport3_x64-uwp/bin/tycam.dll

This bit is required for Windows Store apps.
Found 1 error(s). Please correct the portfile:
    C:\a\1\s\ports\camport3\portfile.cmake

```

The failures on arm-uwp:

```
CMake Error at ports/camport3/portfile.cmake:21 (file):
  file COPY cannot find
  "D:/buildtrees/camport3/src/v1.5.3-d74d139b2a.clean/lib/win/hostapp/arm/tycam.lib":
  File exists.
Call Stack (most recent call first):
  scripts/ports.cmake:79 (include)
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Putting "requires discussion" back on this after further checking it out; couple of thoughts to consider:

1. The upstream git repo seems to indicate that these are an example implementation users are expected to copy and edit themselves; are we sure that putting that here is going to do something users actually want?
2. The `portfile.cmake` is trying to deploy binaries which are checked in to the upstream repo rather than build those binaries. That means things like the compiler requested by the triplet are ignored.

I'm ~95% sure this is probably fine but not sure enough to merge it myself without discussion right now. Will keep you posted.

File: ports/camport3/CONTROL [hunk 5]
```diff
@@ -0,0 +1,5 @@
+Source: camport3
+Version: 1.5.3
+Description: percipio.xyz cameras SDK
+Homepage: https://github.com/percipioxyz/camport3
+Supports: !(osx|uwp|android|wasm32|emscripten|(windows&(static|arm|arm64)))
```
Comment (strega-nil) [inline ports/camport3/CONTROL:5]:
```suggestion
Supports: ((windows & !arm & !arm64 & !static) | linux) !wasm32 
```

**Comment by strega-nil** [comment]
Noting for posterity:

This library is closed source, and the official distribution mechanism is the git repo. We do accept closed source libraries, although we prefer open source mechanisms when possible.

File: ports/camport3/CONTROL (hunk 5 above)
Comment (syp1975) [reply ports/camport3/CONTROL]: I see you are trying to make control file more readable, but some build checks are failing now.

**Comment by syp1975** [comment]
_Supports_ control string is not working. Perhaps I should revert it to [429f7db](https://github.com/microsoft/vcpkg/pull/13759/commits/429f7dbca5b24f099b6121ebd0bc200dde4a71ca)

**Comment by strega-nil** [comment]
@syp1975 nah, it looks like we just missed one case -- this library doesn't support UWP.

File: ports/camport3/CONTROL [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Source: camport3
+Version: 1.5.3
+Description: percipio.xyz cameras SDK
+Homepage: https://github.com/percipioxyz/camport3
+Supports: ((windows & !arm & !arm64 & !static) | linux) & !wasm32
```
Comment (strega-nil) [inline ports/camport3/CONTROL:5]:
```suggestion
Supports: ((windows & !uwp & !arm & !arm64 & !static) | linux) & !wasm32
```

File: ports/camport3/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,82 @@
+if(VCPKG_TARGET_IS_WINDOWS)
+    vcpkg_fail_port_install(ON_LIBRARY_LINKAGE "static")
```
Comment (strega-nil) [inline ports/camport3/portfile.cmake:2]:
```suggestion
    vcpkg_fail_port_install(ON_LIBRARY_LINKAGE "static")
    vcpkg_fail_port_install(ON_TARGET "uwp")
```

**Comment by strega-nil** [comment]
Alright! Thanks so much @syp1975 :)

**Pull request merged**